reinstall-cli/src/config/mod.rs

```diff
@@ -0,0 +1,72 @@
+use anyhow::{bail, ensure, Context, Result};
+use clap::Parser;
+use itertools::Itertools;
+use serde_json::Value;
+
+mod cli;
+
+#[derive(Debug)]
+pub(crate) struct ReinstallConfig {
+    pub(crate) bootc_image: String,
+}
+
+impl ReinstallConfig {
+    pub fn parse_from_cli(cli: cli::Cli) -> Result<Self> {
+        Ok(Self {
+            bootc_image: cli.bootc_image,
+        })
+    }
```

Comment:
why returning `Result<Self>`  instead of just `Self` ?

Reply:
Oops, just leftover from copy pasting from [my previous project where this could fail due to JSON serialization of env::args()](https://github.com/rh-ecosystem-edge/recert/blob/e9428cc982b13bce78f366a628680188086a1797/src/config.rs#L453)

Which come to think of would be nice feature to have here as well, if we wanna leave any trace of the reinstallation configuration, it would be nice to have the raw CLI flags included in that trace